OGM-1229 ORM 5.2 upgrade

gsmet opened the pull request:
Upgrade to ORM 5.2 and Search 5.8.

 * https://hibernate.atlassian.net/browse/OGM-1229
 * https://hibernate.atlassian.net/browse/OGM-1319
 * https://hibernate.atlassian.net/browse/OGM-1320

One thing that we might need to discuss is the removal of NoSQLQuery. It might cause some issues on the user side but relying on NativeQuery (which now has a good name - it's not called SQLQuery anymore) makes it far easier for the integration. If we do want to keep NoSQLQuery, we will need to add a QueryFactory facility to the ORM Session.

Sanne commented:
@gsmet the JTA failure on Neo4j should be solved with the ORM uprade right?

gsmet commented:
@Sanne yes, the JTA issue is fixed. But I have another one with the WF 11 upgrade.

Sanne commented:
Not urgent but for me to have a look it would be useful to rebase it first.

Let me know if I can help with the WF11 upgrade? I tried the current branch but it's attempting to load a non-existing org.hibernate:hibernate-orm-modules:zip:5.2.11.Final with classifier *wildfly-10-dist* so I guess we're not fully in synch

gsmet commented:
Yes, I upgraded to Final, then noticed the WF11 upgrade, then had to upgrade to Search 5.8 and so on :).

I'm working on fixing (hopefully) last issue.

Anyway, we shouldn't merge it for now, I think. Better apply it when we initialize 5.3.

Sanne commented:
> I'm working on fixing (hopefully) last issue.
cool

> Anyway, we shouldn't merge it for now, I think. Better apply it when we initialize 5.3.
Sure, I'd still like to have a peek though :) Also feel free to identify some improvements which could be extracted and merged already.

gsmet commented:
@Sanne rebased!

Sanne commented:
Assigning to @emmanuelbernard as it's just some maintenance, a trivial minor upgrade :D

gsmet commented:
Jenkins, retest this please

Sanne commented on core/src/main/java/org/hibernate/ogm/massindex/impl/OgmMassIndexer.java:
```diff
@@ -171,23 +173,24 @@ private void atLeastOneValidation(int numberOfThreads) {
 	 * @return a new set of entities
 	 */
-	private static Set<Class<?>> toRootEntities(ExtendedSearchIntegrator searchFactoryImplementor, Class<?>... selection) {
+	private static IndexedTypeSet toRootEntities(ExtendedSearchIntegrator extendedIntegrator, Class<?>... selection) {
```
Just to let you know I noticed this, and yes this is going to be much better with the next round of the Search SPI

gsmet commented:
Tried to align the versions with WildFly 11 but I had an issue with Narayana: https://developer.jboss.org/message/976535#976535 .

Worst case scenario, we can keep an older version for the tests but let's see if we can figure it out with the help of the Narayana people.

gsmet commented:
The issue with Narayana was fixed thanks to Tom Jenkinson. I rebased the branch onto latest master.

gsmet commented:
Rebased and upgraded Search to 5.8.1.Final.

gsmet commented:
Rebased. Removed the commits related to the WF 11 upgrade as they are already included in master.

@DavideD I fixed the arbitrary wildfly-10-dist classifier, it's now using the property again.

DavideD commented:
Cool

tduehr commented:
Will OGM start keeping pace with ORM once this is pulled? I'm working on a project that requires support for both.

Sanne commented:
@tduehr yes that's the intention. In practice OGM will always release after ORM, as we need to release that first - but  we're definitely reorganizing and working on better tooling to make sure the gap is minimal.

gsmet commented:
@tduehr btw, the idea is to release an OGM 5.3 shortly after 5.2 to support ORM 5.2.

tduehr commented:
Fantastic! That’ll mean I can stop messing with Kundera!

> On Jan 18, 2018, at 6:19 AM, Guillaume Smet <notifications@github.com> wrote:
> 
> @tduehr btw, the idea is to release an OGM 5.3 shortly after 5.2 to support ORM 5.2.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

gsmet commented:
Just rebased one more time.

gsmet commented:
Jenkins retest this please

DavideD commented on core/src/main/java/org/hibernate/ogm/hibernatecore/impl/OgmSessionFactoryImpl.java:
```diff
@@ -83,4 +93,75 @@ public Reference getReference() throws NamingException {
+	@SuppressWarnings("rawtypes")
+	@Override
+	public EntityGraph findEntityGraphByName(String name) {
+		throw new IllegalStateException( "Hibernate OGM does not support entity graphs" );
```
No need to change this but I'm surprised you used `IllegalStateException` instead of `UnsupportedOperationException`.

gsmet replied:
IIRC, there were preexisting methods using exactly that so I chose to be consistent.

DavideD commented:
Jenkins, rebuild this please

gsmet commented:
@DavideD just force pushed a fix. I didn't have the infinispan-embedded project in my workspace so I didn't see the compilation issue with the new added test. Should be fine now.

DavideD commented:
Merged https://github.com/hibernate/hibernate-ogm/pull/872

Thanks

gsmet commented:
@DavideD thanks! Glad to see this one finally in.
